openedx/core/djangoapps/user_api/tests/test_views.py

```diff
@@ -1405,6 +1467,46 @@ def test_register_with_profile_info(self):
         self.assertEqual(account_settings["goals"], self.GOALS)
         self.assertEqual(account_settings["country"], self.COUNTRY)
 
+    @override_settings(REGISTRATION_EXTENSION_FORM='openedx.core.djangoapps.user_api.tests.test_helpers.TestCaseForm')
+    @mock.patch('openedx.core.djangoapps.user_api.tests.test_helpers.TestCaseForm.DUMMY_STORAGE', new_callable=dict)
+    @mock.patch(
+        'openedx.core.djangoapps.user_api.tests.test_helpers.DummyRegistrationExtensionModel',
+    )
+    def test_with_extended_form(self, dummy_model, storage_dict):
+        dummy_model_instance = mock.Mock()
+        dummy_model.return_value = dummy_model_instance
+        # Create a new registration
+        response = self.client.post(self.url, {
+            "email": self.EMAIL,
+            "name": self.NAME,
+            "username": self.USERNAME,
+            "password": self.PASSWORD,
+            "honor_code": "true",
+            "favorite_movie": "Inception",
+            "favorite_editor": "cat",
+        })
+        self.assertHttpOK(response)
+        self.assertIn(settings.EDXMKTG_LOGGED_IN_COOKIE_NAME, self.client.cookies)
+        self.assertIn(settings.EDXMKTG_USER_INFO_COOKIE_NAME, self.client.cookies)
+
+        user = User.objects.get(username=self.USERNAME)
+        request = RequestFactory().get('/url')
+        request.user = user
+        account_settings = get_account_settings(request)
+
+        self.assertEqual(self.USERNAME, account_settings["username"])
+        self.assertEqual(self.EMAIL, account_settings["email"])
+        self.assertFalse(account_settings["is_active"])
+        self.assertEqual(self.NAME, account_settings["name"])
+
+        self.assertEqual(storage_dict, {'favorite_movie': "Inception", "favorite_editor": "cat"})
```

Comment:
Just to make this test easier to read for future developers, it might be good to explicitly assert that `storage_dict == {}` before the call to the registration endpoint.